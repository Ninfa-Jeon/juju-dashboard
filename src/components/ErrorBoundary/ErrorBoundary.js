import React, { Component } from "react";

export default class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }
  static getDerivedStateFromError(error) {
    return {
      hasError: true,
      error
    };
  }

  componentDidCatch(error, info) {
    // Log to Sentry when enabled
    console.log(error, info);
  }

  render() {
    if (this.state.hasError) {
      // Render error notification
      return (
        <div className="p-notification--negative">
          <p className="p-notification__response">
            <span className="p-notification__status">Error: </span>Something has
            went wrong. If this issue persists,{" "}
            <a href="https://github.com/canonical-web-and-design/jaas-dashboard/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc">
              please raise an issue on Github.
            </a>
          </p>
        </div>
      );
    }

    return this.props.children;
  }
}
