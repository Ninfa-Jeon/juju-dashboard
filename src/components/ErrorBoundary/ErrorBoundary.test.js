import React from "react";
import ErrorBoundary from "./ErrorBoundary";

import { shallow } from "enzyme";

describe("Error Boundary", () => {
  it("generates a error message when an error is caught", () => {
    const wrapper = shallow(<ErrorBoundary />);
    wrapper.setState({
      hasError: true
    });
    expect(wrapper.text()).toEqual(
      "Error: Something has gone wrong. If this issue persists, please raise an issue on GitHub."
    );
  });
});
