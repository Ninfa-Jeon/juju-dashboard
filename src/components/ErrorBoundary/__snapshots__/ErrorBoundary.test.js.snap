// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary renders without crashing and matches snapshot 1`] = `
<div
  className="p-notification--negative"
>
  <p
    className="p-notification__response"
  >
    <span
      className="p-notification__status"
    >
      Error: 
    </span>
    Something has gone wrong. If this issue persists,
     
    <a
      href="https://github.com/canonical-web-and-design/jaas-dashboard/issues/new?assignees=&labels=&template=bug_report.md&title="
    >
      please raise an issue on GitHub.
    </a>
  </p>
</div>
`;
